utils/modeler/modeler.go

```diff
@@ -0,0 +1,107 @@
+package modeler
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+// Modeler is a utility for populating an arbitrary model's fields with values from a
+// map[string]string.
+type Modeler struct {
+	prefix string
+	tag    string
+}
+
+// NewModeler returns a pointer to a new Modeler, confgiured with the provided map key prefix and
+// struct tag key.
+func NewModeler(prefix string, tag string) *Modeler {
+	return &Modeler{prefix: prefix, tag: tag}
+}
+
+// MapToModel populates the provided model with values from the provided map.
```

Comment:
would be nice to have some docs here on some of the errors to expect. not blocking for this PR, however